include/rpt/rpt_cell_reconstruction.h

```diff
@@ -0,0 +1,169 @@
+
+*
+* Authors: Audrey Collard-Daigneault, Polytechnique Montreal, 2020-
+*/
+
+#ifndef lethe_rpt_cell_reconstruction_h
+#define lethe_rpt_cell_reconstruction_h
+
+/**
+ * This class allows cell position reconstruction of unknown particle positions
+ * with counts value for many detector. It finds the center position of the best
+ * cell that its vertices counts range contains the counts value of unknown
+ * particle positions.
+ */
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/grid/tria.h>
+
+#include <rpt/detector.h>
+#include <rpt/parameters_rpt.h>
+#include <rpt/radioactive_particle.h>
+#include <rpt/rpt_calculating_parameters.h>
+
+#include <map>
+#include <vector>
+
+using namespace dealii;
+
+template <int dim>
+class RPTCellReconstruction
+{
+public:
+  /**
+   * @brief Constructor for the RPTCellReconstruction
+   *
+   * @param detectors Vector of all detector object
+   *
+   * @param rpt_parameters All parameters for the RPT
+   *
+   */
+  RPTCellReconstruction(RPTCalculatingParameters &rpt_parameters);
+
+  /**
+   * @brief Set up grid and find positions for every unknown particle positions.
+   */
+  void
+  execute_cell_reconstruction();
+
+  /**
+   * @brief Create grid of the reactor vessel (cylinder).
+   */
+  void
+  create_grid();
+
+  /**
+   * @brief Read text file for detector positions and assign them to detector
+   * objects
+   */
+  void
+  assign_detector_positions();
+
+  /**
+   * @brief Calculate counts at vertices for the coarse mesh.
+   */
+  void
+  set_coarse_mesh_counts();
+
+  /**
+   * @brief Find the position of the best cell that contains the unknown particle
+   * position
+   *
+   * @param particle_reconstruction_counts Counts the the current unknown particle
+   * position for every detector
+   */
+  void
+  find_unknown_position(std::vector<double> &particle_reconstruction_counts);
+
+  /**
+   * @brief Find the vertices position of coarse mesh vertices or the candidate
+   * cells
+   *
+   * @param level Level of the current cells of the mesh
+   *
+   * @param parent_cell_indexes List of the parent cells which are candidate.
+   *                            Is {-1} by default when executing coarse mesh
+   *                            (no parent cells)
+   */
+  void
+  find_vertices_positions(unsigned int     level,
+                          std::vector<int> parent_cell_indexes = {-1});
+  /**
+   * @brief Find the cell candidates indexes for the unknown particle positions
+   * cells
+   *
+   * @param level Level of the current cells of the mesh
+   *
+   * @param parent_cell_indexes List of the parent cells which are candidate.
+   *                            Is {-1} by default when executing coarse mesh
+   *                            (no parent cells)
+   */
+  std::vector<int>
+  find_cells(unsigned int         level,
+             std::vector<double> &particle_reconstruction_counts,
+             std::vector<int>     parent_cell_indexes = {-1});
```

Comment:
Why int? cells are of type types::global_cell_index  (see https://www.dealii.org/current/doxygen/deal.II/classCellAccessor.html)

Reply:
Indexes the way I use it seem to be int.
![image](https://user-images.githubusercontent.com/71884806/128366948-43455510-1c75-43e6-ac51-6a2ae9aee544.png)
![image](https://user-images.githubusercontent.com/71884806/128367349-120b8973-ab34-454e-af5b-ab576fe4fa2a.png)